pkg/security/apis/security/v1/defaults.go

```diff
@@ -1,107 +1,18 @@
 package v1
 
 import (
+	//v1 "github.com/openshift/api/security/v1"
+	//"github.com/openshift/origin/pkg/cmd/openshift-kube-apiserver/admission/customresourcevalidation/securitycontextconstraints"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
-
-	"github.com/openshift/api/security/v1"
-	sccutil "github.com/openshift/origin/pkg/security/securitycontextconstraints/util"
 )
 
 func AddDefaultingFuncs(scheme *runtime.Scheme) error {
 	RegisterDefaults(scheme)
-	scheme.AddTypeDefaultingFunc(&v1.SecurityContextConstraints{}, func(obj interface{}) { SetDefaults_SCC(obj.(*v1.SecurityContextConstraints)) })
-	return nil
-}
-
-// Default SCCs for new fields.  FSGroup and SupplementalGroups are
-// set to the RunAsAny strategy if they are unset on the scc.
-func SetDefaults_SCC(scc *v1.SecurityContextConstraints) {
-	if len(scc.FSGroup.Type) == 0 {
-		scc.FSGroup.Type = v1.FSGroupStrategyRunAsAny
-	}
-	if len(scc.SupplementalGroups.Type) == 0 {
-		scc.SupplementalGroups.Type = v1.SupplementalGroupsStrategyRunAsAny
-	}
-
-	if scc.Users == nil {
-		scc.Users = []string{}
-	}
-	if scc.Groups == nil {
-		scc.Groups = []string{}
-	}
-
-	var defaultAllowedVolumes sets.String
-	switch {
-	case scc.Volumes == nil:
-		// assume a nil volume slice is allowing everything for backwards compatibility
-		defaultAllowedVolumes = sets.NewString(string(v1.FSTypeAll))
-
-	case len(scc.Volumes) == 0 && scc.AllowHostDirVolumePlugin:
-		// an empty volume slice means "allow no volumes", but the boolean fields will always take precedence.
-		defaultAllowedVolumes = sets.NewString(string(v1.FSTypeHostPath))
-
-	case len(scc.Volumes) == 0 && !scc.AllowHostDirVolumePlugin:
-		// an empty volume slice means "allow no volumes", but cannot be persisted in protobuf.
-		// convert this to volumes:["none"]
-		defaultAllowedVolumes = sets.NewString(string(v1.FSTypeNone))
-
-	default:
-		// defaults the volume slice of the SCC.
-		// In order to support old clients the boolean fields will always take precedence.
-		defaultAllowedVolumes = fsTypeToStringSet(scc.Volumes)
-	}
-
-	if scc.AllowHostDirVolumePlugin {
-		// if already allowing all then there is no reason to add
-		if !defaultAllowedVolumes.Has(string(v1.FSTypeAll)) {
-			defaultAllowedVolumes.Insert(string(v1.FSTypeHostPath))
-		}
-	} else {
-		// we should only default all volumes if the SCC came in with FSTypeAll or we defaulted it
-		// otherwise we should only change the volumes slice to ensure that it does not conflict with
-		// the AllowHostDirVolumePlugin setting
-		shouldDefaultAllVolumes := defaultAllowedVolumes.Has(string(v1.FSTypeAll))
+	/*
+		scheme.AddTypeDefaultingFunc(&v1.SecurityContextConstraints{}, func(obj interface{}) {
```

Comment:
I don't know if adding this will get the unit tests to pass, but adding it does create a circular dependency on the o-kas SCC package